pytorch_lightning/core/lightning.py

```diff
@@ -1585,21 +1589,34 @@ def to_torchscript(
             This LightningModule as a torchscript, regardless of whether file_path is
             defined or not.
         """
-
         mode = self.training
-        with torch.no_grad():
-            if method == 'script':
-                torchscript_module = torch.jit.script(self.eval(), **kwargs)
-            elif method == 'trace':
-                # if no example inputs are provided, try to see if model has example_input_array set
-                if example_inputs is None:
-                    example_inputs = self.example_input_array
-                # automatically send example inputs to the right device and use trace
-                example_inputs = self.transfer_batch_to_device(example_inputs, device=self.device)
-                torchscript_module = torch.jit.trace(func=self.eval(), example_inputs=example_inputs, **kwargs)
-            else:
-                raise ValueError(f"The 'method' parameter only supports 'script' or 'trace', but value given was:"
-                                 f"{method}")
+
+        if method == 'script':
+            torchscript_module = torch.jit.script(self.eval(), **kwargs)
+        elif method == 'trace':
+            # if no example inputs are provided, try to see if model has example_input_array set
+            if example_inputs is None:
+                if self.example_input_array is None:
+                    raise ValueError(
+                        'Choosing method=`trace` requires either `example_inputs`'
+                        ' or `model.example_input_array` to be defined'
+                    )
+                example_inputs = self.example_input_array
+
+            if isinstance(example_inputs, Tensor):
+                example_inputs = (example_inputs,)
+            elif not (isinstance(example_inputs, tuple) and all(isinstance(inp, Tensor) for inp in example_inputs)):
+                raise ValueError(
+                    "Could not export to torchscript since example_inputs is neither a Tensor nor tuple of Tensors"
+                )
+
+            # automatically send example inputs to the right device and use trace
+            example_inputs = self.transfer_batch_to_device(example_inputs, device=self.device)
```

Comment:
This part of the code is very similar in the two methods, so maybe it could be refactored?

Reply:
yep it can be. I'll try.

Reply:
how about exporting both these methods into a separate class `LightningExportModule` or something else??